### PR TITLE
43 add post endpoints to redux createapi

### PR DESCRIPTION
**Pull request by DLiSWE**

### CODE REVIEW

**Review by Matt-Tan15 — DISMISSED** [review]
looks good

**Review by raylu511 — APPROVED** [review]

File: src/features/posts/models/Posts.model.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "content" | "media" | "categoryName"
+>;
+
+export type CreatePostRequestModel = Omit<BasePostModel, "id">;
+export type DeletePostRequestModel = Pick<BasePostModel, "id" | "userId">;
+
+export type LikePostModel = Pick<BasePostModel, "id" | "userId">;
+export type DeletePostLikeModel = Pick<BasePostModel, "id" | "userId">;
+
+export type PostIdBodyProps = {
```
Comment (raylu511) [inline src/features/posts/models/Posts.model.ts:22]: Should we also be able to update the media of the post?
Also the type name is weird.

Comment (raylu511) [inline src/features/posts/models/Posts.model.ts:22]: I would name this `UpdatePostRequestModel` instead.
And also type `postId `as a number.

**Review by raylu511 — CHANGES_REQUESTED** [review]
Did you test all of the endpoints out?

**Comment by DLiSWE** [comment]
I did

File: src/features/auth/components/LoginForm.tsx [hunk 2]
```diff
@@ -6,12 +6,14 @@ import { LoginUserModel } from "../models/User.model";
 import { useForm } from "shared/hooks";
 import { useLoginMutation } from "../services/authApiSlice";
 import { loginUser } from "../authSlice";
+import { useCreatePostMutation } from "features/posts/services/postApiSlice";
```
Comment (Matt-Tan15) [inline src/features/auth/components/LoginForm.tsx:9]: what is this used for on the LoginForm?

Comment (DLiSWE) [reply src/features/auth/components/LoginForm.tsx:9]: you right

**Review by Matt-Tan15 — APPROVED** [review]

File: src/shared/components/SharePost.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {
   Typography,
 } from "@mui/material";
 import { useForm } from "../hooks";
-import { BaseCreatePostmodel } from "../types/SharePostTypes";
+import { BasePostRequestModel } from "features/posts/models/Posts.model";
 
 type SharePostProps = {
```
Comment (Matt-Tan15) [inline src/shared/components/SharePost.tsx:15]: should we change this to SharePostPropType?

**Pull request closed without merging**